test/lightning/install_schemas_test.exs

```diff
@@ -0,0 +1,174 @@
+defmodule Lightning.InstallSchemasTest do
+  use ExUnit.Case, async: false
+  use Mimic
+
+  alias Mix.Tasks.Lightning.InstallSchemas
+
+  @request_options [recv_timeout: 15_000, pool: :default]
+  @ok_200 {:ok, 200, "headers", :client}
+  @ok_400 {:ok, 400, "headers", :client}
+
+  describe "install_schemas mix task" do
+    setup do
+      stub(:hackney)
+
+      :ok
+    end
+
+    test "run success" do
+      expect(:hackney, :request, fn
+        :get,
+        "https://registry.npmjs.org/-/user/openfn/package",
+        [],
+        "",
+        @request_options ->
+          @ok_200
+      end)
+
+      expect(:hackney, :body, fn :client, _timeout ->
+        {:ok,
+         ~s({"@openfn/language-primero": "write","@openfn/language-asana": "write", "@openfn/language-common": "write"})}
+      end)
+
+      expect(:hackney, :request, fn
+        :get,
+        "https://cdn.jsdelivr.net/npm/@openfn/language-asana/configuration-schema.json",
+        [],
+        "",
+        @request_options ->
+          @ok_200
+      end)
+
+      expect(:hackney, :body, fn :client, _timeout ->
+        {:ok, ~s({"name": "language-asana"})}
+      end)
+
+      expect(:hackney, :request, fn
+        :get,
+        "https://cdn.jsdelivr.net/npm/@openfn/language-primero/configuration-schema.json",
+        [],
+        "",
+        @request_options ->
+          @ok_200
+      end)
+
+      expect(:hackney, :body, fn :client, _timeout ->
+        {:ok, ~s({"name": "language-primero"})}
+      end)
+
+      File
+      |> expect(:rm_rf, fn _ -> nil end)
+      |> expect(:mkdir_p, fn _ -> nil end)
+      |> expect(:open!, fn "priv/schemas/asana.json", [:write] -> nil end)
+      |> expect(:open!, fn "priv/schemas/primero.json", [:write] -> nil end)
+      |> expect(:close, 2, fn _ -> nil end)
+
+      IO
+      |> expect(:binwrite, fn _, ~s({"name": "language-asana"}) -> nil end)
+      |> expect(:binwrite, fn _, ~s({"name": "language-primero"}) -> nil end)
```

Comment:
I appreciate the effort in trying to stub both the HTTP and filesystem calls, but I think it's a little overkill.

Next time, or in the future I'd propose having the location of where to persist them be stubbed to `/tmp/priv/schemas` - and while stubbing the http calls, test that the files were actually written. It's probably easier imo.

Reply:
Totally agree. Here we're just ensuring that the expected content will be written.